### ISSUE TITLE
Modal: Focus trap sometimes broken

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#1753]
Description  
---
When opening a modal it detects all focusable elements inside. This can cause problems if focusable (web-component) elements are not detected or the content with focusable elements is dynamically changed. So that the focus trap is not working as expected.

Out of scope
---
- We don't want to provide a stable / performant solution for cases to avoid from UX / technical perspective, e.g. heavy nested forms / tables within a modal or iframe

Notes
---
- Ionic Solution: https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/overlays.ts#L38
- Material Solution: https://mui.com/components/trap-focus/ & https://github.com/mui/material-ui/blob/master/packages/mui-base/src/Unstable_TrapFocus/Unstable_TrapFocus.js
- We want to offer a solution that detects all relevant, focusable elements incl. PDS ShadowDOM Components

Here are some Slack community feedbacks which covers this issue:
- https://porschedev.slack.com/archives/CT7AVHTTQ/p1643636452439859
- https://porschedev.slack.com/archives/CT7AVHTTQ/p1643368526351379
- https://porschedev.slack.com/archives/CT7AVHTTQ/p1630682380033000

Acceptance Criteria
---
- [ ] The modal is able to detection of focusable elements in ShadowDOM
- [ ] The modal is able to detect changes of focusable elements within the content
 
Tasks
---
- [ ] Extend technical docu with focus trap explanation and for limitations and recommendations
